### PR TITLE
Prevents delete of instance groups list item controlplane and default

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses #10396. `controlplane` instance group is part of every build where as the `default` instance group is part of Openshit/Kubernetes install.
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION
Screen Shot 2021-06-10 at 10 06 42 AM

### CODE REVIEW

File: awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx [hunk 1]
```diff
@@ -25,7 +25,11 @@ const QS_CONFIG = getQSConfig('instance-group', {
   page_size: 20,
 });
 
-function modifyInstanceGroups(items = []) {
+function modifyInstanceGroups(
+  items = [],
+  defaultControlPlan,
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx]:
```suggestion
  defaultControlPlane,
```

File: awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx [hunk 2]
```diff
@@ -36,7 +40,7 @@ function modifyInstanceGroups(items = []) {
         },
       },
     };
-    if (clonedItem.name === 'tower') {
+    if (clonedItem.name === (defaultControlPlan || defaultExecution)) {
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx]:
```suggestion
    if (clonedItem.name === (defaultControlPlane || defaultExecution)) {
```

Comment (jakemcdermott) [reply awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx]: (and a few other places)

**Review by jakemcdermott — CHANGES_REQUESTED** [review]
lgtm, but we'll want to rename the variable from `plan` -> `plane` and _possibly_ move the settings request into it's own request hook (see comments)

File: awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx [hunk 3]
```diff
@@ -57,18 +61,32 @@ function InstanceGroupList() {
...
       searchableKeys,
+      defaultControlPlan,
+      defaultExecution,
     },
   } = useRequest(
     useCallback(async () => {
       const params = parseQueryString(QS_CONFIG, location.search);
 
-      const [response, responseActions] = await Promise.all([
+      const [
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx]: @AlexSCorey Will the new settings request run once on initialization or every time the user applies a search filter? If it's running for every search, we'll want to:
* put this into it's own `useRequest` hook instead to make sure we only call it once.
* or add some kind of conditional to skip the settings request if we already have the data
* or some other approach, if you prefer. The important thing is to not the settings endpoint more than needed.

**Review by jakemcdermott — APPROVED** [review]

**Comment by nixocio** [comment]
@AlexSCorey the changes look great.

~However on this branch I tried to create a new instance group and the following error was raised.~

image

~I did not see the error on devel.~

That error seems to be on my setup.

**Comment by akus062381** [comment]
@AlexSCorey - I saw the comment @jakemcdermott made, so I checked- and when I tried to create a new instance group, I saw the following error:
image

The feature of not being able to delete these two instance groups works great!

**Comment by AlexSCorey** [comment]
recheck

**Review by nixocio — APPROVED** [review]
Last issue reported was fixed.

**Comment by nixocio** [comment]
I triggered an e2e test run.

**Review by akus062381 — APPROVED** [review]
great!

**Comment by AlexSCorey** [comment]
regate

**Comment by AlexSCorey** [comment]
recheck

**Pull request closed without merging**